src/diffusers/models/prior_transformer.py

```diff
@@ -91,25 +116,34 @@ def __init__(
             ]
         )
 
+        if has_pre_norm:
+            self.norm_in = nn.LayerNorm(inner_dim)
+        else:
+            self.norm_in = None
+
         self.norm_out = nn.LayerNorm(inner_dim)
-        self.proj_to_clip_embeddings = nn.Linear(inner_dim, embedding_dim)
+
+        self.proj_to_clip_embeddings = nn.Linear(inner_dim, out_dim)
 
         causal_attention_mask = torch.full(
             [num_embeddings + additional_embeddings, num_embeddings + additional_embeddings], -10000.0
         )
         causal_attention_mask.triu_(1)
         causal_attention_mask = causal_attention_mask[None, ...]
         self.register_buffer("causal_attention_mask", causal_attention_mask, persistent=False)
-
-        self.clip_mean = nn.Parameter(torch.zeros(1, embedding_dim))
-        self.clip_std = nn.Parameter(torch.zeros(1, embedding_dim))
+        if has_post_process:
+            self.clip_mean = nn.Parameter(torch.zeros(1, clip_embedding_dim))
```

Comment:
This doesn't look backwards compatible. `clip_embedding_dim` should by default be set to `embedding_dim` in the `__init__(...)` above.

Reply:
I think it's already set to `embedding_dim` by default, but let me know if I understood wrong